tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -35,411 +35,414 @@
-    f2 = epyccel(cosh_phrase)
+    f2 = epyccel(cosh_phrase, language = language)
     x = rand()
     y = rand()
     assert(isclose(f2(x,y), cosh_phrase(x,y), rtol=1e-15, atol=1e-15))
 
-def test_tanh_call():
+def test_tanh_call(language):
     @types('real')
     def tanh_call(x):
         from numpy import tanh
         return tanh(x)
 
-    f1 = epyccel(tanh_call)
+    f1 = epyccel(tanh_call, language = language)
     x = rand()
     assert(isclose(f1(x), tanh_call(x), rtol=1e-15, atol=1e-15))
 
-def test_tanh_phrase():
+def test_tanh_phrase(language):
     @types('real','real')
     def tanh_phrase(x,y):
         from numpy import tanh
         a = tanh(x)+tanh(y)
         return a
 
-    f2 = epyccel(tanh_phrase)
+    f2 = epyccel(tanh_phrase, language = language)
     x = rand()
     y = rand()
     assert(isclose(f2(x,y), tanh_phrase(x,y), rtol=1e-15, atol=1e-15))
 
-def test_arctan2_call():
+def test_arctan2_call(language):
     @types('real','real')
     def arctan2_call(x,y):
         from numpy import arctan2
         return arctan2(x,y)
 
-    f1 = epyccel(arctan2_call)
+    f1 = epyccel(arctan2_call, language = language)
     x = rand()
     y = rand()
     assert(isclose(f1(x,y), arctan2_call(x,y), rtol=1e-15, atol=1e-15))
 
-def test_arctan2_phrase():
+def test_arctan2_phrase(language):
     @types('real','real','real')
     def arctan2_phrase(x,y,z):
         from numpy import arctan2
         a = arctan2(x,y)+arctan2(x,z)
         return a
 
-    f2 = epyccel(arctan2_phrase)
+    f2 = epyccel(arctan2_phrase, language = language)
     x = -rand()
     y = rand()
     z = rand()
     assert(isclose(f2(x,y,z), arctan2_phrase(x,y,z), rtol=1e-15, atol=1e-15))
 
-def test_sqrt_call():
+def test_sqrt_call(language):
     @types('real')
     def sqrt_call(x):
         from numpy import sqrt
         return sqrt(x)
 
-    f1 = epyccel(sqrt_call)
+    f1 = epyccel(sqrt_call, language = language)
     x = rand()
     assert(isclose(f1(x), sqrt_call(x), rtol=1e-15, atol=1e-15))
 
-def test_sqrt_phrase():
+def test_sqrt_phrase(language):
     @types('real','real')
     def sqrt_phrase(x,y):
         from numpy import sqrt
         a = sqrt(x)*sqrt(y)
         return a
 
-    f2 = epyccel(sqrt_phrase)
+    f2 = epyccel(sqrt_phrase, language = language)
     x = rand()
     y = rand()
     assert(isclose(f2(x,y), sqrt_phrase(x,y), rtol=1e-15, atol=1e-15))
 
-
-def test_sqrt_return_type():
+def test_sqrt_return_type_r(language):
     @types('real')
     def sqrt_return_type_real(x):
         from numpy import sqrt
         a = sqrt(x)
         return a
+
+    f1 = epyccel(sqrt_return_type_real, language = language)
+    x = rand()
+    assert(isclose(f1(x), sqrt_return_type_real(x), rtol=1e-15, atol=1e-15))
+    assert(type(f1(x)) == type(sqrt_return_type_real(x).item())) # pylint: disable=unidiomatic-typecheck
+
+def test_sqrt_return_type_c(language):
+    if (language == 'c'):
+        pytest.xfail(reason="complex type not implimented yet")
```

Comment:
This should be a decorator of the function. See above for the syntax to xfail for one language only